src/main.cpp

```diff
@@ -3,3 +3,127 @@
  */
 
 #include "Arduino.h"
+#include <SimpleFOC.h>
+
+//  Encoder(int encA, int encB , int cpr, int index)
+//  - encA, encB    - encoder A and B pins
+//  - ppr           - impulses per rotation  (cpr=ppr*4)
+//  - index pin     - (optional input)
+Encoder sensor = Encoder(2, 3, 2048); //will we have hardware or software interrupts?
```

Comment:
I'm still not sure if we use the built-in Encoder class, or if we have to create a new sensor class for the Pico PIO. Probably the latter, I think the built-in encoder class will set up a hardware interrupt on the encoder pins which will overwhelm the processor at 2048ppr.

Reply:
It does allow us to setup software interrupts instead as an option too. However, I agree that we may have to create a new class anyway.

Reply:
Software interrupts may be even slower than hardware interrupts. The Pico PIO makes it so that we don't need to interrupt the processor every time we get an encoder pulse, the PIO records that and then we can just ask the PIO for the current position.

Reply:
Oh right. I'll look into creating this new sensor class and implement it in a follow-on PR. I'll focus this PR on setting up the generic parameters and configurations.

Reply:
Let me know if you need a hand with that!